### ISSUE TITLE
[HOLD for payment 2023-08-10] [$1000] Web - Task - Assignee tooltip does not show display name, but 'Share somewhere' tooltip does

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#23632]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to staging.new.expensify.com
2. Tap + icon > Assign task > Enter title > Next.
3. Select an assignee with display name and 'Share somewhere' user with display name.
4. Hover over 'Assignee' avatar'.
5. Hover over 'Share somwhere' avatar.

## Expected Result:
Tooltip for both 'Assignee' and 'Share somewhere' avatar shows user display name.

## Actual Result:
Assignee tooltip does not show assignee display name, but 'Share somewhere' tooltip does.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.45-3
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/0a30d131-550d-4aa6-898d-f2a0d746064e

**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~017d5e761a395a2035</li>
        <li>Upwork Job ID: 1684295448780296192</li>
        2023-07-26<li>Automatic offers: </li>
<ul>
<li>c3024 | Contributor | 25797428</li>
</ul></li>
    </ul>
</details>

**Comment by daanish-shaikh** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
In Assign Task process at Confirm Task step, display name of the user on the tooltip is visible on 'Share somewhere' but not on assignee.

### What is the root cause of that problem?
Data passed in the `icon` prop of `MenuItem` component in src/pages/tasks/NewTaskPage.js is missing `id` field which is required [here](https://github.com/Expensify/App/blob/main/src/components/UserDetailsTooltip/index.web.js#L20).

### What changes do you think we should make in order to solve the problem?
We'll need to modify `getAssignee()` function to fix the missing `id` field in `assignee` state data.

Add  `, id: details.accountID`, [here](https://github.com/Expensify/App/blob/main/src/libs/actions/Task.js#L584). 
After which the function will look like this.
```
function getAssignee(details) {
    if (!details) {
        return {
            icons: [],
            displayName: '',
            subtitle: '',
        };
    }
    const source = UserUtils.getAvatar(lodashGet(details, 'avatar', ''), lodashGet(details, 'accountID', -1));
    return {
        icons: [{source, type: 'avatar', name: details.login, id: details.accountID}],
        displayName: details.displayName,
        subtitle: details.login,
    };
}
```

**Comment by Santhosh-Sellavel** [comment]
@thienlnam 

@c3024's [proposal](https://github.com/Expensify/App/issues/23632#issuecomment-1651734963) looks good to me.

I like both solutions main and alternative.

Preferably the alternative, utils come handy
> We may add a getIcons function to UserUtils similar to the getIcons in ReportUtils and move the logic of icons to it. 

Let me know your thoughts.

**Comment by Santhosh-Sellavel** [comment]
@daanish-shaikh Please don't post duplicate proposals, thanks!

**Comment by thienlnam** [comment]
Yup looks good, let's try to DRY up the code by doing the alternative as well in implementation

**Comment by daanish-shaikh** [comment]
> @daanish-shaikh Please don't post duplicate proposals, thanks!

Sorry guys, wasn't intentional. I carelessly posted the proposal without checking @c3024's proposal. Will be careful in the future.

**Comment by Santhosh-Sellavel** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web & Desktop </summary>

https://github.com/Expensify/App/assets/85645967/c91c7043-c12e-4b81-91d4-add7caf07679

</details>

<details>
<summary>Mobile Web - Chrome</summary>

![screenshot-1690834355335](https://github.com/Expensify/App/assets/85645967/1ee58d46-2030-42b0-9686-b7ade08567b2)

</details>

<details>
<summary>Mobile Web - Safari</summary>

![Simulator Screenshot - iPhone 14 - 2023-08-01 at 01 40 33](https://github.com/Expensify/App/assets/85645967/be1d1984-85be-419e-a31d-a50a84525cc8)

</details>

<details>
<summary>iOS</summary>

![Simulator Screenshot - iPhone 14 - 2023-08-01 at 01 38 13](https://github.com/Expensify/App/assets/85645967/7c66f363-fcc0-4c6c-935a-bde403885a75)

</details>

<details>
<summary>Android</summary>

![screenshot-1690834055656](https://github.com/Expensify/App/assets/85645967/9a3f944a-b5bc-43ff-bdc2-8d7198af798c)

</details>

**Comment by garrettmknight** [comment]
@Santhosh-Sellavel once you knock out the checklist I'll pay this one out.

@c3024 paying via Upwork now.

**Comment by Santhosh-Sellavel** [comment]
I'll collect via new dot once I'm done with the checklist!

**Comment by Santhosh-Sellavel** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

* [x]  [@Santhosh-Sellavel] The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/20276
* [x]  [@Santhosh-Sellavel] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/20276#issuecomment-1673839117
* [x]  [@Santhosh-Sellavel] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: Should have been while testing.
* [x]  [@Santhosh-Sellavel] Determine if we should create a regression test for this bug. Not sure
* [x]  [@Santhosh-Sellavel] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
* [ ]  [@garrettmknight] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by Santhosh-Sellavel** [comment]
**Bug:** Task - Assignee tooltip does not show display name

### Regression Test Steps
1. Click FAB and click on assign task
2. Give a task name and click next.
3. Select an assignee with a display name different from their login
4. Hover on the avatar to see that display name is visible in the tooltip

👍  or 👎

**Comment by Santhosh-Sellavel** [comment]
@garrettmknight Can you leave a payment summary message, thanks!

**Comment by garrettmknight** [comment]
**Summarizing payments for this issue:**

- Issue reporter: Applause - N/A
- Contributor: @c3024 - $1500 paid via Upwork
- Contributor+: @Santhosh-Sellavel - $1500

Upwork job [here](https://www.upwork.com/jobs/~017d5e761a395a2035)

**Comment by Santhosh-Sellavel** [comment]
Requested on ND

**Comment by garrettmknight** [comment]
@JmillsExpensify can you confirm if you've paid this one?

**Comment by JmillsExpensify** [comment]
Reviewed the details for @Santhosh-Sellavel. Approved for payment in NewDot based on the BZ summary above.
